src/chatd.cpp

```diff
@@ -423,6 +423,8 @@ Promise<void> Connection::reconnect(const std::string& url)
                     mConnectPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                     mLoginPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                 }
+
+                return err;
```

Comment:
No need here, the promise handler chain is not used by anything, so it doesn't matter whether the finally returned promise is. Actually, returning the error will cause an "Unhandled promise fail" warning to be logged by the promise lib.
However, I see another bug here - if the error is something else than an SDK error, it will be silently ignored. You need to change the fail() handle to something like:
```
    int errType =  (err.type() == ERRTYPE_MEGASDK) ? WS_ERRTYPE_DNS : err.type();
     mConnectPromise.reject(err.msg(), err.code(), errtype);
     mLoginPromise.reject(err.msg(), err.code(), errtype);